controllers/topology/current_state_test.go

```diff
@@ -209,14 +183,16 @@ func TestGetCurrentState(t *testing.T) {
 			wantErr: true,
 		},
 		{
-			name: "Cluster with ControlPlane reference and with ControlPlane Infrastructure, but no InfrastructureCluster",
+			name: "Should read a partial Cluster (with ControlPlane and control plane infrastructure machine template, but without Infrastructure Cluster and workers)",
 			cluster: testtypes.NewClusterBuilder(metav1.NamespaceDefault, "cluster1").
+				// No Infrastructure Cluster!
```

Comment:
```suggestion
				// No InfrastructureCluster!
```
Last nit of this type though there's more to fix up. My understanding (from the last time this code was reviewed 😄 ) is that it's best practice to keep the type names as is in order to prevent drift and ensure consistency in what's being referred to.